lib/engine/step/g_1860/track.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+require_relative 'tracker'
+
+module Engine
+  module Step
+    module G1860
+      class Track < Base
+        include Tracker
+        ACTIONS = %w[lay_tile pass].freeze
+
+        def actions(entity)
+          return [] if entity.company? || !can_lay_tile?(entity)
+
+          entity == current_entity ? ACTIONS : []
+        end
+
+        def description
+          'Lay Track'
+        end
+
+        def pass_description
+          @acted ? 'Done (Track)' : 'Skip (Track)'
+        end
+
+        def process_lay_tile(action)
+          lay_tile_action(action)
+          pass! unless can_lay_tile?(action.entity)
+        end
+
+        def reachable_path?(entity, path)
+          max_distance = @game.biggest_train(entity)
+          return false if max_distance.zero?
+
+          path_distances = @game.path_distances(entity)
+          path_distances[path] <= max_distance
+        end
+
+        def reachable_node?(entity, node)
+          max_distance = @game.biggest_train(entity)
+          return false if max_distance.zero?
+
+          node_distances = @game.node_distances(entity)
+          return false unless node_distances[node]
+
+          if node.city?
+            node_distances[node] < max_distance
+          else
+            node_distances[node] <= max_distance
+          end
+        end
+
+        def reachable_hex?(entity, hex)
+          max_distance = @game.biggest_train(entity)
+          return false if max_distance.zero?
+
+          node_distances = @game.node_distances(entity)
+          path_distances = @game.path_distances(entity)
+          return false unless hex.tile.paths.any? { |p| path_distances[p] }
+
+          # tile currently on network
+          if hex.tile.nodes.any?
+            # tile has a city/town/halt
+            hex.tile.nodes.each do |tile_node|
+              nd = node_distances[tile_node] if node_distances[tile_node]
+              if tile_node.city? || tile_node.offboard?
+                return true if nd && nd < max_distance
+              elsif nd && nd <= max_distance
+                return true
+              end
+            end
+          else
+            # tile is just track
+            # Assumption: 1860 has no tiles that have track that doesn't connect to a node on the same tile
+            hex.tile.paths.each do |tile_path|
+              pd = path_distances[tile_path] if path_distances[tile_path]
+              return true if pd && pd <= max_distance
+            end
+          end
+
+          false
+        end
+
+        def available_hex(entity, hex)
+          return false unless (neighbors = @game.graph.connected_hexes(entity)[hex])
```

Comment:
does this get hit on loading?

Reply:
No. Same answer as hex_neighbors. This is no different from the base Tracker module.